chain/client/src/metrics.rs

```diff
@@ -637,3 +637,35 @@ pub(crate) static PARTIAL_WITNESS_CACHE_SIZE: LazyLock<Gauge> = LazyLock::new(||
     )
     .unwrap()
 });
+
+pub(crate) static RECEIVE_WITNESS_ACCESSED_CONTRACT_CODES_TIME: LazyLock<HistogramVec> =
+    LazyLock::new(|| {
+        try_create_histogram_vec(
+            "near_receive_witness_accessed_contract_codes_time",
+            "Time it takes to retrieve missing contract codes",
+            &["shard_id"],
+            Some(linear_buckets(0.025, 0.025, 40).unwrap()),
+        )
+        .unwrap()
+    });
+
+pub(crate) static WITNESS_ACCESSED_CONTRACT_CODES_DELAY: LazyLock<HistogramVec> =
+    LazyLock::new(|| {
+        try_create_histogram_vec(
+            "near_witness_accessed_contract_codes_delay",
+            "Delay in witness processing caused by waiting for accessed contract codes",
```

Comment:
just to clarify, this is after all sufficient parts are collected (so without code distribution we would otherwise start validating) and before getting code, right?

Reply:
yes, exactly